api/python/tests/test_cli.py

```diff
@@ -54,14 +54,49 @@ def test_push_force(self):
 
             main.main(('push', '--dir', dir_path, name, '--force'))
 
-            mocked_package_class.browse.assert_called_once_with(name, None)
+            mocked_package_class.browse.assert_not_called()
+            mocked_package_class.assert_called_once_with()
+            mocked_package = mocked_package_class.return_value
+            mocked_package.set_dir.assert_called_once_with('.', dir_path)
+            mocked_package.push.assert_called_once_with(
+                name, registry=None, dest=None, message=None, workflow=..., force=True, dedupe=False
+            )
+
+    def test_push_request_put(self):
+        name = 'test/name'
+        dir_path = 'test/dir/path'
+
+        with patch_package_class as mocked_package_class:
+            mocked_package_class.browse.side_effect = FileNotFoundError()
+
+            main.main(('push', '--dir', dir_path, name, '--request', 'put'))
+
+            mocked_package_class.browse.assert_not_called()
             mocked_package_class.assert_called_once_with()
             mocked_package = mocked_package_class.return_value
-            mocked_package.set_dir.assert_called_once_with('.', dir_path, meta=None)
+            mocked_package.set_dir.assert_called_once_with('.', dir_path)
             mocked_package.push.assert_called_once_with(
                 name, registry=None, dest=None, message=None, workflow=..., force=True, dedupe=False
             )
 
+    def test_push_request_patch(self):
+        name = 'test/name'
+        dir_path = 'test/dir/path'
+        registry = 's3://my-bucket'
+
+        with patch_package_class as mocked_package_class:
+            mocked_package_class.browse.side_effect = FileNotFoundError()
+
+            main.main(('push', '--dir', dir_path, name, '--request', 'patch', '--registry', registry))
+
+            mocked_package_class.browse.assert_called_once_with(name, registry)
+            mocked_package_class.assert_called_once_with()
+            mocked_package = mocked_package_class.return_value
+            mocked_package.set_dir.assert_called_once_with('.', dir_path)
+            mocked_package.push.assert_called_once_with(
+                name, registry=registry, dest=None, message=None, workflow=..., force=False, dedupe=False
+            )
+
```

Comment:
I have simplistic integration tests that confirm the CLI is calling the correct API.  I still don't know the right way to ensure we are providing the correct semantics. Any suggestions?